include/libdnf/rpm/package_query.hpp

```diff
@@ -364,40 +574,41 @@ class PackageQuery : public PackageSet {
 
     PackageQuery & filter_available();
 
-    /// @replaces libdnf/sack/query.hpp:method:addFilter(int keyname, int cmp_type, int match) - cmp_type = HY_PKG_UPGRADES
+    // @replaces libdnf/sack/query.hpp:method:addFilter(int keyname, int cmp_type, int match) - cmp_type = HY_PKG_UPGRADES
     PackageQuery & filter_upgrades();
 
-    /// @replaces libdnf/sack/query.hpp:method:addFilter(int keyname, int cmp_type, int match) - cmp_type = HY_PKG_DOWNGRADES
+    // @replaces libdnf/sack/query.hpp:method:addFilter(int keyname, int cmp_type, int match) - cmp_type = HY_PKG_DOWNGRADES
     PackageQuery & filter_downgrades();
 
-    /// @replaces libdnf/sack/query.hpp:method:addFilter(int keyname, int cmp_type, int match) - cmp_type = HY_PKG_UPGRADABLE
+    // @replaces libdnf/sack/query.hpp:method:addFilter(int keyname, int cmp_type, int match) - cmp_type = HY_PKG_UPGRADABLE
     PackageQuery & filter_upgradable();
 
-    /// @replaces libdnf/sack/query.hpp:method:addFilter(int keyname, int cmp_type, int match) - cmp_type = HY_PKG_DOWNGRADABLE
+    // @replaces libdnf/sack/query.hpp:method:addFilter(int keyname, int cmp_type, int match) - cmp_type = HY_PKG_DOWNGRADABLE
     PackageQuery & filter_downgradable();
 
-    /// @brief Keep number of latest versions of packages. When negative limit is used it removes a number of latest versions.
-    ///
-    /// @version 1.0.0
-    /// @param limit Default value is 1
-    /// @returns Self
+    /// Filter packages by grouping them by `name` and `arch` and keeping up to `limit` number of them from each group.
```

Comment:
What about?
```/// Filter packages by grouping them by `name`, `arch` and version and keeping up to `limit` number of them from each group.```
Without version it looks like that it returns number of architectures

Reply:
I don't think the sentence you have proposed is correct.
Version (and Epoch and Release) is not part of the grouping.
Or did I miss anything? In that case, could you provide several examples how the filter works?

Reply:
I suggest that the solution is somewhere in the middle. It groups according `name` and `arch`, And inside each group it groups according to versions (EVR).  Then for each `name` and `arch` group it keeps number of EVR groups according to `limit` value.

Reply:
@j-mracek then FWIW @dmach's formulation seems to be the correct one.

Reply:
In case when multiple packages with the same NEVRA are present, then the formulation is misleading.
`A-1-1.noarch`
`A-1-1.noarch`
`A-1-2.noarch`
`A-1-2.noarch`
With `limit=1` it will return `A-1-2.noarch` and `A-1-2.noarch`. 
From `and keeping up to `limit` number of them from each group` I expect to return only one package from `name.arch` group whem `limit=1`.

Reply:
Suggesting a minor change in `and keeping up to limit number of them from each group` to `and keeping up to limit number of versions from each group`

Reply:
I did my best to rephrase the docstring to make it more clear. Please take a look.